src/clib/pio_darray.c

```diff
@@ -323,13 +333,14 @@ int PIOc_write_darray(const int ncid, const int vid, const int ioid,
         }
         else
         {
+	    /* Move to end of list. */
             while(wmb->next && wmb->ioid != -(ioid))
                 if (wmb->next)
                     wmb = wmb->next;
         }
     }
 
-    /* ?? */
```

Comment:
the write multi buffer wmulti_buffer is the cache on compute nodes that will collect and store multiple variables 
before sending them to the io nodes.   Aggregating variables in this way leads to a considerable savings in communication expense.  Variables in the wmb array must have the same decomposition and base data size and we also need to keep track of whether each is a recordvar (has an unlimited dimension) or not.